core/src/lib.rs

```diff
@@ -183,20 +190,26 @@ where
 
 /// Creates a JSON-RPC 2.0 request to the given method with the given parameters.
 fn serialize_request<P>(
-    id: u64,
-    method: &str,
+    id: Id,
+    method: String,
     params: P,
 ) -> ::std::result::Result<Vec<u8>, serde_json::error::Error>
 where
     P: serde::Serialize,
 {
-    let mut request_map = serde_json::Map::new();
-    request_map.insert(String::from("jsonrpc"), JsonValue::from("2.0"));
-    request_map.insert(String::from("id"), JsonValue::from(id));
-    request_map.insert(String::from("method"), JsonValue::from(method));
-    request_map.insert(String::from("params"), serde_json::to_value(params)?);
-
-    serde_json::to_vec(&JsonValue::Object(request_map))
+    let serialized_params = match serde_json::to_value(params)? {
+        JsonValue::Null => None,
+        JsonValue::Array(vec) => Some(Params::Array(vec)),
+        JsonValue::Object(obj) => Some(Params::Map(obj)),
+        value => Some(Params::Array(vec![value])),
+    };
+    let method_call = MethodCall {
+        jsonrpc: Some(Version::V2),
+        method: method.to_owned(),
```

Comment:
Seems to be owned already.

Reply:
Ah. Correct. A leftover from earlier when `method` was a `&str`. Fixing that.